### PR TITLE
Merge upstream changes

### PR DESCRIPTION
**Pull request by kurayama**

### CODE REVIEW

File: lib/reader.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ var delegate = utils.delegate;
  * Expose `Reader`.
  */
 
-module.exports = Reader;
+exports = module.exports = Reader;
```
Comment (kurayama) [inline lib/reader.js]: ¯\\\_(ツ)_/¯

**Comment by rplopes** [comment]
For the record, upstream changes not included in this PR:

```diff
diff --git c/lib/reader.js w/lib/reader.js
index 207c108..61e07d5 100644
--- c/lib/reader.js
+++ w/lib/reader.js
@@ -19,7 +19,7 @@ var delegate = utils.delegate;
  * Expose `Reader`.
  */

-exports = module.exports = Reader;
+module.exports = Reader;

 /**
  * Initialize a new Reader.
diff --git c/test/acceptance/reader.js w/test/acceptance/reader.js
index 35871ed..7e64795 100644
--- c/test/acceptance/reader.js
+++ w/test/acceptance/reader.js
@@ -3,8 +3,6 @@ var utils = require('../utils');
 var assert = require('assert');
 var nsq = require('../..');
 var uid = require('uid');
-var Reader = require('../../lib/reader');
-

 describe('Reader', function(){
   var topic = uid();
@@ -49,7 +47,7 @@ describe('Reader', function(){

         sub.on('done', done);
       });
-    });
+    })

     describe('with .nsqlookupd addresses', function(){
       it('should subscribe to messages', function(done){
@@ -83,7 +81,7 @@ describe('Reader', function(){
           done();
         });
       })
-    });
+    })

     it('should discard messages after the max attempts', function(done){
       var pub = nsq.writer();
```

**Review by rplopes — COMMENTED** [review]
Squash some of those commits, since they're not consistent on their own and we're not following upstream's history anymore anyway?

Other than that, LGTM, the missing changes are IMO unnecessary.

**Comment by kurayama** [comment]
Squashed the commits and made a single one since most of them were irrelevant and too granular.

**Review by rplopes — APPROVED** [review]

**Pull request merged**